File: packages/camera/camera/lib/src/camera_barcode.dart
/// CameraBarcode
class CameraBarcode {
  /// text
  final String text;

  /// format
  final String format;

  /// errorDescription;
  final String? errorDescription;

  /// constructor
  CameraBarcode({
    required this.text,
    required this.format,
    required this.errorDescription,
  });

  /// fromJson
  CameraBarcode.fromJson(Map<dynamic, dynamic> json)
      : text = json['text'],
        format = json['format'],
        errorDescription = json['errorDescription'];

  @override
  String toString() {
    return 'CameraBarcode{text: $text, format: $format, errorDescription: $errorDescription}';
  }
}

File: packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/types/CameraBarcode.java
package io.flutter.plugins.camera.types;

import java.util.HashMap;

public class CameraBarcode {
    final String text;
    final String format;
    final String errorDescription;

    public CameraBarcode(String text, String format, String errorDescription) {
        this.text = text;
        this.format = format;
        this.errorDescription = errorDescription;
    }

    public HashMap<String, Object> getMap() {
        return new HashMap<String, Object>() {
            {
                put("text", text != null ? text : "");
                put("format", format != null ? format : "");
                put("errorDescription", errorDescription);
            }
        };
    }

}
